feedback-mailer/feedback-mailer-from-sheet.gs

```diff
@@ -0,0 +1,474 @@
+`;
+
+function onInstall(e) {
+  // Without this, user will have to reload the sheet after installing.
+  onOpen(e);
+}
+
+function onOpen(e) {
+  SpreadsheetApp.getUi()
+  .createMenu(MENU_TITLE)
+  .addItem(SEND_MAILS, "sendMailsToStudents")
+  .addItem(GENERATE_HEADERS, "generateHeaders")
+  .addItem(MENU_HELP, "help")
+  .addToUi();
+}
+
+function help() {
+  showInfo(TITLE_HELP, MSG_HELP);
+}
+
+function showInfo(title, body) {
+  let ui = SpreadsheetApp.getUi();
+  ui.alert(title, body, ui.ButtonSet.OK);
+}
+
+function logBeginDelimiter() {
+  console.log("BEGIN: 10x---------------10x");
+}
+
+function logEndDelimiter() {
+  console.log("END: 10x---------------10x");
+}
+
+function generateHeaders() {
+  const workSheet = SpreadsheetApp.getActiveSheet();
+  workSheet.getRange(1, INDEX_SCORE + 1).setValue(SCORE_HEADER_TITLE);
+  workSheet.getRange(1, INDEX_MAIL_STATUS + 1).setValue(MAIL_STATUS_HEADER_TITLE);
+  showInfo(TITLE_SUCCESS, MSG_HEADERS_GENERATED);
+}
+
+/**
+ * Ask the user if they want to use their current email to sent the emails.
+ */
+function confirmSendingMail(email) {
+  let question = `Current email belongs to the following account. Do you want to use this account?
+
+  Id: ${email}
+  `;
+
+  let ui = SpreadsheetApp.getUi();
+  let userResponse = ui.alert("Please confirm", question, ui.ButtonSet.YES_NO);
+  if (userResponse === ui.Button.YES) {
+    return true;
+  }
+  logBeginDelimiter();
+  console.log(MSG_MAIL_REJECTED);
+  logEndDelimiter();
+  return false;
+}
+
+function initSetup() {
+  let result = {
+    success: false,
+    activeSheet: null,
+    feedbackData: null
+  };
+
+  result.activeSheet = SpreadsheetApp.getActiveSheet();
+  result.feedbackData = sheetData(result.activeSheet);
+  if (!result.feedbackData) {
+    showInfo(TITLE_ERROR, ERROR_CODE_COULD_NOT_GET_DATA);
+    return result;
+  }
+  result.success = true;
+  return result;
+}
+
+function sheetData(sheet) {
+  try {
+    let dataRange = sheet.getRange(`A${DATA_BEGIN_ROW}:${DATA_END_COLUMN}${sheet.getLastRow()}`);
+    return dataRange.getDisplayValues();
+  } catch (error) {
+    logBeginDelimiter();
+    console.log(ERROR_CODE_COULD_NOT_GET_DATA);
+    console.log(error);
+    logEndDelimiter();
+    return null;
+  }
+}
+
+function validateEmail(email) {
+  try {
+    return email.endsWith("@gmail.com") || email.endsWith("@the10xacademy.com");
+  } catch (error) {
+    return false;
+  }
+}
+
+function sendEmail(studentEmail, mailBody) {
+  let status = "";
+  let errorMsg = "";
+  if (!validateEmail(studentEmail)) {
+    status = MAIL_STATUS_FAILED;
+    errorMsg = `Invalid email: ${studentEmail}`;
+    return { status , errorMsg };
+  }
+  try {
+    let htmlFile = HtmlService.createTemplate(mailBody);
+    let htmlText = htmlFile.evaluate().getContent();
+    GmailApp.sendEmail(studentEmail, MAIL_SUBJECT, mailBody, { name: OWNER_NAME , htmlBody : htmlText } );
```

Comment:
`mailBody` is alread an HTML text. Why should you apply `HTMLService` on top of it?

Reply:
Previously I was converting the `mailBody` from normal text to `htmlText` as, I was saving the `mailBody` into sheets.
This part was left out from previous version.
Now We are not doing that, as we are directly using `htmlText`, This is corrected in next commit.